test/teiserver/game/balancer_server_test.exs

```diff
@@ -36,7 +83,147 @@ defmodule Teiserver.Game.BalancerServerTest do
+      GenServer.call(
+        start_link_results,
+        {:make_balance, team_count, [], players},
+        5000
+      )
+
+    state3 = GenServer.call(start_link_results, :report_state, 5000)
+    dbg(state3)
+
+    assert second_balance_pass_hash_reuse.hash == first_balance_pass.hash
+
+    # users = [
+    #   make_user(%{
+    #     "name" => "user"+ExULID.ULID.generate(),
+    #     "email" => ExULID.ULID.generate()+"@example.com",
+    #     "permissions" => []
+    #   }),
+    #   make_user(%{
+    #     "name" => "user2",
+    #     "email" => "user2-test542684@example.com",
+    #     "permissions" => []
+    #   }),
+    #   make_user(%{
+    #     "name" => "user3",
+    #     "email" => "user3-test542684@example.com",
+    #     "permissions" => []
+    #   }),
+    #   make_user(%{
+    #     "name" => "user4",
+    #     "email" => "user4-test542684@example.com",
+    #     "permissions" => []
+    #   }),
+    #   make_user(%{
+    #     "name" => "user5",
+    #     "email" => "user5-test542684@example.com",
+    #     "permissions" => []
+    #   })
+    # ]
+
+    #### TODO
+
+    # third_balance_pass_with_more_players =
+    #   GenServer.call(
+    #     start_link_results,
+    #     {:make_balance, team_count, [], create_fake_players(160)},
+    #     5000
+    #   )
+
+    # dbg(third_balance_pass_with_more_players)
+    # assert second_balance_pass_hash_reuse.hash != third_balance_pass_with_more_players.hash
+  end
+
+  test "get_balance_mode works with a hash" do
+    team_count = 2
+
+    arbitrary_rank_for_all_test_users = 10
+
+    players = make_users_with_ranks([10, 20, 30, 40])
+
+    {:ok, start_link_results} = BalancerServer.start_link(data: %{lobby_id: 1})
+
+    starting_state = GenServer.call(start_link_results, :report_state, 5000)
+
+    first_balance_pass =
+      GenServer.call(
+        start_link_results,
+        {:make_balance, team_count, [], players},
+        5000
+      )
+
+    get_balance_mode_returns_value =
+      GenServer.call(
+        start_link_results,
+        :get_balance_mode,
+        5000
+      )
+
+    refute get_balance_mode_returns_value == nil
+  end
+
+  test "get_current_balance works with a hash" do
+    team_count = 2
+
+    arbitrary_rank_for_all_test_users = 10
+
+    players = make_users_with_ranks([10, 20, 30, 40])
+
+    {:ok, start_link_results} = BalancerServer.start_link(data: %{lobby_id: 1})
+
+    starting_state = GenServer.call(start_link_results, :report_state, 5000)
+
+    first_balance_pass =
+      GenServer.call(
+        start_link_results,
+        {:make_balance, team_count, [], players},
+        5000
+      )
+
+    get_balance_mode_returns_value =
+      GenServer.call(
+        start_link_results,
+        :get_current_balance,
+        5000
+      )
+
+    refute get_balance_mode_returns_value == nil
+  end
+
   defp create_fake_players(count) do
-    1..count |> Enum.map(fn _ -> %{} end)
+    1..count |> Enum.map(fn iter -> %{iter: iter} end)
```

Comment:
I'll probably revert this function to the original form; I thought I could make it serve double-duty but the needs of the test harness evolved beyond simple tweaks